### PR TITLE
Repos loaded accordingly to sat version

### LINKED ISSUE #3698: The jobs should use it's own respective sat-tools repos
Our Robottelo automation should pick up dynamically depending upon various pipeline/releases the SatTools versions.
1) For Sat6.1.z
We need to enable and sync SatTools RH Repo 6.1 version 
2) For Sat6.2.z
We need to enable and sync SatTools 6.2 custom repos
For custom repos, we can make use of this config param, https://github.com/SatelliteQE/robottelo-ci/blob/master/scripts/satellite6-automation.sh#L84 from `robottelo.properties` file.
This value is populated from `Satellite 6 Repos URLs Environment Variables` configfile.
3) For Sat6.2 GA  
We need to enable and sync SatTools RH Repo 6.2 version 
4) For Sat6.3 
We need to enable and sync SatTools 6.3 custom repos
For custom repos, we can make use of this config param, https://github.com/SatelliteQE/robottelo-ci/blob/master/scripts/satellite6-automation.sh#L84 from `robottelo.properties` file.
This value is populated from `Satellite 6 Repos URLs Environment Variables` configfile.

### PR DESCRIPTION
**Pull request by renzon**
closes #3698

Refactoring:
Host information unified on host_info

### CODE REVIEW

**Comment by renzon** [comment]
@oshtaier been more specific than issue discation, alll ContentViewTestCase tests are going to fail on sat 6.2 becase [create_rhel_contet](https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/cli/test_contentview.py#L92) uses a non existing REPO on 6.2: 'rhva6'. So we need add it on 6.2 or change the key used here. I am search a little bit more to see if there is more impact in another tests.

**Comment by renzon** [comment]
Result for robottelo unit tests:

```console
/home/renzo/PycharmProjects/venvs/robottelo/bin/python /home/renzo/bin/pycharm-2016.2/helpers/pycharm/pytestrunner.py -p pytest_teamcity /home/renzo/PycharmProjects/robottelo/tests/robottelo
Testing started at 12:54 PM ...
============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/renzo/PycharmProjects/robottelo, inifile: 
plugins: cov-2.3.0, xdist-1.14
collected 165 items

test_api_utils.py ..
test_cli.py .....
test_config_casts.py ..............
test_config_settings.py .....
test_datafactory.py .....
test_decorators.py ...........................................
test_hammer.py .........
test_helpers.py ...........
test_host_info.py ................
test_ssh.py ...........
test_ui.py .....
test_ui_locators.py ........................
test_vm.py ......
decorators/test_host.py .........

========================== 165 passed in 1.66 seconds ==========================
```

**Comment by oshtaier** [comment]
please fix:
```
tests/robottelo/test_host_info.py:2: in <module>
    from mock.mock import call
E   ImportError: No module named 'mock'
===================== 149 passed, 1 error in 5.01 seconds ======================
ERROR: InvocationError: '/home/travis/build/SatelliteQE/robottelo/.tox/py34/bin/py.test --cov --cov-config=.coveragerc tests/robottelo'
___________________________________ summary ____________________________________
ERROR:   py34: commands failed
```

**Comment by oshtaier** [comment]
probably, something like:
```
 if six.PY2:
      import mock
  else:
      from unittest import mock
```

**Comment by renzon** [comment]
```console
 48 tests deselected by '-k ContentViewTestCase and test_positive_add_rh_repo_by_id' 
============= 1 failed, 1 passed, 48 deselected in 319.38 seconds ==============
```
In fact test_positive_add_rh_repo_by_id passes on test and it uses REPOS variable. So it seems be working as before

File: robottelo/host_info.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            r'[^\d]*(?P<version>\d(\.\d){1})'
+        )
+        result = re.search(version_re, version_description)
+        if result:
+            host_sat_version = result.group('version')
+            return host_sat_version, ssh_result
+
+    return 'Not Available', ssh_result
+
+
```
Comment (rochacbruno) [inline robottelo/host_info.py]: We can get rid of the __main__ block below, as we already have tests for this it is not useful no?

**Comment by rochacbruno** [comment]
ACK 
(pending non blocker comment)

File: robottelo/host_info.py (hunk 1 above)
Comment (renzon) [inline robottelo/host_info.py]: No, I just forgot to remove it

**Comment by renzon** [comment]
@oshtaier fixed. Let me know if you find any issue while using it on you issue.

**Comment by oshtaier** [comment]
ACK

**Comment by oshtaier** [comment]
@renzon 
Re-verified my test case with your code. Everything passed as it should (6.2 repos were selected as they should)

**Comment by sghai** [comment]
Nice Job @renzon

File: robottelo/host_info.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        if result:
+            host_os_version = 'RHEL{}'.format(result.group('version'))
+            LOGGER.debug('Host version: {}'.format(host_os_version))
+            return host_os_version
+
+    LOGGER.warning('Host version not available: {!r}'.format(cmd))
+    return 'Not Available'
+
+
+_SAT_6_2_VERSION_COMMAND = (
```
Comment (rplevka) [inline robottelo/host_info.py]: I don't think we want this here as it utilizes downstream package which does not ship with Katello (in upstream)

Comment (rochacbruno) [inline robottelo/host_info.py]: @rplevka so in upstream is there another path? so we can put the `grep pattern` and also the `version file path` in settings variable.

**Comment by lpramuk** [comment]
@rochacbruno Don't we have separate branches for 6.1 and for 6.2 ? 

I don't get why renzo cares about Sat Versions?

File: robottelo/host_info.py (hunk 2 above)
Comment (rplevka) [inline robottelo/host_info.py]: @rochacbruno there's no 'satellite' at all in upstream, we only have a katello version. The whole concept of 'satellite' version is wrong here. Not sure what is this used for, but we have a different branch for Sat6.2 and Sat6.1, so I think the code merged to master should aim for 6.2 only.

**Comment by rochacbruno** [comment]
@lpramuk this feature was requested in #3698 take a look. Maybe @sghai knows why branch was not an option for this issue.

**Pull request merged**